test/modules.coffee

```diff
@@ -0,0 +1,492 @@
+  input = """
+    import { foo } from 'lib'
+    bar = 5"""
+  output = """
+    var bar;
+
+    import {
+      foo
+    } from 'lib';
+
+    bar = 5;"""
+  eq toJS(input), output
+
+test "variables can be assigned before and after an import", ->
+  input = """
+    foo = 5
+    import { bar } from 'lib'
+    baz = 7"""
+  output = """
+    var baz, foo;
+
+    foo = 5;
+
+    import {
+      bar
+    } from 'lib';
+
+    baz = 7;"""
+  eq toJS(input), output
+
+# Export statements
+
+test "export named members within an object", ->
+  input = "export { foo, bar }"
+  output = """
+    export {
+      foo,
+      bar
+    };"""
+  eq toJS(input), output
+
+test "export named members as aliases, within an object", ->
+  input = "export { foo as bar, baz as qux }"
+  output = """
+    export {
+      foo as bar,
+      baz as qux
+    };"""
+  eq toJS(input), output
+
+test "export default string", ->
+  input = "export default 'foo'"
+  output = "export default 'foo';"
+  eq toJS(input), output
+
+test "export default number", ->
+  input = "export default 5"
+  output = "export default 5;"
+  eq toJS(input), output
+
+test "export default object", ->
+  input = "export default { foo: 'bar', baz: 'qux' }"
+  output = """
+    export default {
+      foo: 'bar',
+      baz: 'qux'
+    };"""
+  eq toJS(input), output
+
+test "export default assignment expression", ->
+  input = "export default foo = 'bar'"
+  output = """
+    var foo;
+
+    export default foo = 'bar';"""
+  eq toJS(input), output
+
+test "export assignment expression", ->
+  input = "export foo = 'bar'"
+  output = "export var foo = 'bar';"
+  eq toJS(input), output
+
+test "export default function", ->
+  input = "export default ->"
+  output = "export default function() {};"
+  eq toJS(input), output
+
+test "export default multiline function", ->
+  input = """
+    export default (foo) ->
+      console.log foo"""
+  output = """
+    export default function(foo) {
+      return console.log(foo);
+    };"""
+  eq toJS(input), output
+
+test "export assignment function", ->
+  input = """
+    export foo = (bar) ->
+      console.log bar"""
+  output = """
+    export var foo = function(bar) {
+      return console.log(bar);
+    };"""
+  eq toJS(input), output
+
+test "export assignment function which contains assignments in its body", ->
+  input = """
+    export foo = (bar) ->
+      baz = '!'
+      console.log bar + baz"""
+  output = """
+    export var foo = function(bar) {
+      var baz;
+      baz = '!';
+      return console.log(bar + baz);
+    };"""
+  eq toJS(input), output
+
+test "export default predefined function", ->
+  input = """
+    foo = (bar) ->
+      console.log bar
+    export default foo"""
+  output = """
+    var foo;
+
+    foo = function(bar) {
+      return console.log(bar);
+    };
+
+    export default foo;"""
+  eq toJS(input), output
+
+# Uncomment this test once ES2015+ `class` support is added
```

Comment:
We do support it (see line 353). This is a test that checks that `export default class foo` _outputs_ `export default class foo`, e.g. it outputs the ES2015 `class` keyword. Currently `class` gets output as `function`, which is what we’re testing for below; but eventually we want `class` (or some other keyword like `esclass`) to output as `class`, and when that day comes we should uncomment this test.